src/mobile/components/Orders/index.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { useIntl } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { CloseIcon } from '../../../assets/images/CloseIcon';
+import { TabPanel } from '../../../components';
+import { useUserOrdersHistoryFetch } from '../../../hooks';
+import {
+    ordersCancelAllFetch,
+    ordersHistoryCancelFetch,
+    selectOrdersHistory,
+    selectShouldFetchCancelAll,
+    selectShouldFetchCancelSingle,
+} from '../../../modules';
+import { OrdersItem } from './OrdersItem';
+
+const userOrdersHistoryTabs = ['open', 'all'];
+
+const OrdersComponent = () => {
+    const [currentTabIndex, setCurrentTabIndex] = React.useState(0);
+    const dispatch = useDispatch();
+    const intl = useIntl();
+    const orders = useSelector(selectOrdersHistory);
+    const shouldFetchCancelAll = useSelector(selectShouldFetchCancelAll);
+    const shouldFetchCancelSingle = useSelector(selectShouldFetchCancelSingle);
+    useUserOrdersHistoryFetch(0, userOrdersHistoryTabs[currentTabIndex], 25);
+
+    const handleCancelAllOrders = () => {
+        if (shouldFetchCancelAll) {
+            dispatch(ordersCancelAllFetch());
+        }
+    };
+
+    const handleCancelSingleOrder = (id: number) => () => {
+        if (shouldFetchCancelAll && shouldFetchCancelSingle) {
+            dispatch(ordersHistoryCancelFetch({
+                id,
+                type: userOrdersHistoryTabs[currentTabIndex],
+                list: orders,
+            }));
+        }
+    };
+
+    const renderOptionalHead = () => (
```

Comment:
use `useCallback` for all inner methods inside React.FC components (inside all components)